universal_login/deploy.ts

```diff
@@ -64,164 +79,45 @@ function loadEmail(emailName: Email): Record<string, unknown> {
-  env: Env,
-  template: Template,
-  token: BearerToken
-) {
-  const apiHost = getApiHost(env);
+  const textPromptEndpoint = `/api/v2/prompts/${prompt}/custom-text/${language}`;
 
-  const templateData = loadHTMLTemplate(template);
-  const templateEndpoint = `https://${apiHost}/api/v2/branding/templates/${template}`;
-
-  const templateUpdateRequest = {
-    method: 'PUT',
-    url: templateEndpoint,
-    headers: {
-      authorization: `Bearer ${token.access_token}`,
-      'content-type': 'text/html',
-    },
-    data: templateData,
-  } as AxiosRequestConfig;
-
-  await axios.request(templateUpdateRequest);
+  return instance.put(textPromptEndpoint, promptData);
 }
 
-async function updateEmail(env: Env, email: Email, token: BearerToken) {
-  const apiHost = getApiHost(env);
+function updateLoginPageTemplate(
+  instance: AxiosInstance,
+  template: Template
+): Promise<void> {
+  const templateData = loadHTMLTemplate(template);
+  const templateEndpoint = `/api/v2/branding/templates/${template}`;
 
-  const emailData = loadEmail(email);
-  const emailEndpoint = `https://${apiHost}/api/v2/email-templates/${email}`;
-
-  const templateUpdateRequest = {
-    method: 'PUT',
-    url: emailEndpoint,
-    headers: {
-      authorization: `Bearer ${token.access_token}`,
-      'content-type': 'application/json',
-    },
-    data: emailData,
-  } as AxiosRequestConfig;
-
-  await axios.request(templateUpdateRequest);
+  return instance.put(templateEndpoint, templateData, {
+    headers: { 'content-type': 'text/html' },
+  });
 }
 
-(async () => {
-  try {
-    const env = process.argv[2] as Env;
-    console.log(`Updating environment: ${env}`);
-
-    const token = await getApiToken(env);
-    // The non production rate limit of the API end point is 2 requests a second, this ensures we never hit the limit
-    const delay = () => new Promise((res) => setTimeout(res, 1000));
+async function updateEmail(
+  instance: AxiosInstance,
+  email: Email,
+  config: Record<string, string>
+): Promise<void> {
+  const emailData = loadEmail(email);
+  const emailEndpoint = `/api/v2/email-templates/${email}`;
 
-    for (const prompt of prompts) {
-      await delay();
-      console.log(`Updating ${prompt} prompt`);
-      await updateTextPrompt(env, prompt, token);
+  for (const [key, value] of Object.entries(emailData)) {
+    if (typeof value === 'string') {
+      emailData[key] = applyTemplates(value, config);
     }
+  }
 
-    console.log(`Updating universal-login template`);
-    await updateLoginPageTemplate(env, 'universal-login', token);
-
-    for (const email of emails) {
-      await delay();
-      console.log(`Updating ${email} email`);
-      await updateEmail(env, email, token);
-    }
+  return instance.put(emailEndpoint, emailData);
+}
 
-    console.log('Updates completed successfully');
-  } catch (e) {
-    console.error(e);
-    process.exit(1);
-  }
-})();
+const env = process.argv[2] as Env;
```

Comment:
What sort of error do we throw if `process.argv[2]` is missing? Or it's not an `Env` value?

I'd add a `try … catch` here (or the TS equivalent) that prints a usage string if it's used incorrectly.